src/app/vm/container/vm-tags.container.ts

```diff
@@ -40,14 +40,14 @@ export class VmTagsContainerComponent extends WithUnsubscribe() implements OnIni
 
   public editTag(tagEditAction: TagEditAction) {
     const newTag = {
-      resourceIds: this.vm.id,
-      resourceType: this.vm.resourceType,
+      resourceid: this.vm.id,
+      resourcetype: this.vm.resourceType,
       key: tagEditAction.newTag.key,
       value: tagEditAction.newTag.value
     };
     const newTags = Object.assign([], this.vm.tags)
       .filter(t => tagEditAction.oldTag.key !== t.key);
-    newTags.push(new Tag(newTag));
+    newTags.push(newTag as Tag);
```

Comment:
remove as, add typing to new tags and remove Object.assign
add the same changes for similar code